### ISSUE TITLE
Connection Issue - "Failed To Connect" and Socket Connection error

### ISSUE THREAD

**Issue by brettlandau** [BlueBubblesApp/bluebubbles-server#687]
I first witnessed this happening today. Server has been running fine for a few weeks now.

First, there was an issue sending a message, and I got the notification on my Android device "Could not connect - Your server may be offline!". In the settings panel of the BlueBubbles app on Android I see a socket connection error: 
![Screenshot_20240822-183051](https://github.com/user-attachments/assets/03d786be-8fbf-4e31-ad8a-d8e47c3c3723)
. Strange, so I went to manually reboot my Mac and see if that helped.

When I did the reboot, I noticed that the message that I was trying to send earlier actually did go through (it showed in the Mac Messages app). After reboot, I got the same "Could not connect" notification and socket connection error.

While this was all happening, I could actually continue to send and receive messages as if nothing was wrong. Thought it would fix itself after a few hours but it didn't so I:
- installed BB server 1.9.8: no change
- deleted Android app and reinstalled: this actually made things worse since now I cannot reconnect for long enough to download the messages. After tapping on the Firebase project and entering my server pw, I immediately get the popup "Failed To Connect! Please make sure you are connected to the internet and your server is online!"
- tried BB web app: it starts to download messages (I can see the logs) but then I get the "Failed To Connect!" after a few seconds.

Any ideas what might be going on? Could my ISP be messing things up with the server? Thanks, all help is much appreciated.

**Comment by HenryTheX** [comment]
Exactly the same issue need help. Since yesterday so annoying.

**Comment by DaSnof** [comment]
I was having the same issue, and I think it's related to Cloudflare. I also deleted the app and was getting the same connection error when trying to set it back up.

I took a few minutes and set the server up using port forwarding and the dynamic DNS option just for trial purposes, and I seem to be back up and running as usual again.

**Comment by alexpaluzzi** [comment]
Also started happening to me in the last day. But it still actually works. It just annoyingly keeps giving this notification.

**Comment by brettlandau** [comment]
I wish I hadn’t deleted the app 😅 

Interesting to see I’m not the only one having this issue. I can collect logs if it’s helpful.

**Comment by JeffreyGECH** [comment]
I agree, wish I hadn't deleted what I deleted.  Removed and reinstalled BlueBubbles.  Rebuilt Cloudflare over and over.  Tried different email addresses and reinstalled BlueBubbles over and over and nothing changed.  Also tried 2 user accounts on MAC, and that was worse.

No errors in my BlueBubbles logs.  Everything looks good.  Just can't connect any client to it.  Not Pixel, not any desktop version.  Nothing is working and kinda freaking out.

Now I'm getting this "failed to connect" message.  Found people saying to turn Wi-Fi on and off, but that hasn't helped.
![Screenshot_20240823-113358](https://github.com/user-attachments/assets/718dc816-bf65-4728-a251-6621772e9169)

Any help is appreciated!!

**Comment by tneotia** [comment]
Everyone, please switch your server proxy to Zrok or set up a self-hosted solution like Tailscale. Zrok setup is extremely easy:

```
go to the mac you use as a server
open bluebubbles server app
go to settings
under Connection Settings, Proxy Setup, select "Zrok (Beta)" from the drop down
follow on-screen instructions which are:
type in email, press button
go to your email, click link
choose password
log in
go to "Detail" tab on the Zrok web interface
click the eye to show your token
copy token
paste into bluebubbles server where it says to
click the equivalent of the "ok" button in bluebubbles server 
```

Your server should now show a zrok URL and your client should pick that up.

**Comment by cameronaaron** [comment]
@tneotia I had recommended to Zach that we switch to Zrok by default. Cloudflare tunnels by default both changes a lot and also can be easily deprioritized not to mention that you cannot send large videos. What are your thoughts on making that change

**Comment by IceBlueLugia** [comment]
> Everyone, please switch your server proxy to Zrok or set up a self-hosted solution like Tailscale. Zrok setup is extremely easy:
> 
> ```
> go to the mac you use as a server
> open bluebubbles server app
> go to settings
> under Connection Settings, Proxy Setup, select "Zrok (Beta)" from the drop down
> follow on-screen instructions which are:
> type in email, press button
> go to your email, click link
> choose password
> log in
> go to "Detail" tab on the Zrok web interface
> click the eye to show your token
> copy token
> paste into bluebubbles server where it says to
> click the equivalent of the "ok" button in bluebubbles server 
> ```
> 
> Your server should now show a zrok URL and your client should pick that up.

This seemed to work for a bit but during syncing it disconnected. Do you know if this still works?
